nessie/src/main/java/org/apache/iceberg/nessie/NessieViewOperations.java

```diff
@@ -0,0 +1,164 @@
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+import org.projectnessie.model.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NessieViewOperations extends BaseViewOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(NessieViewOperations.class);
+
+  private final NessieIcebergClient client;
+  private final ContentKey key;
+  private final FileIO fileIO;
+  private final Map<String, String> catalogOptions;
+  private IcebergView icebergView;
+
+  NessieViewOperations(
+      ContentKey key,
+      NessieIcebergClient client,
+      FileIO fileIO,
+      Map<String, String> catalogOptions) {
+    this.key = key;
+    this.client = client;
+    this.fileIO = fileIO;
+    this.catalogOptions = catalogOptions;
+  }
+
+  @Override
+  public void doRefresh() {
+    try {
+      client.refresh();
+    } catch (NessieNotFoundException e) {
+      throw new RuntimeException(
+          String.format(
+              "Failed to refresh as ref '%s' is no longer valid.", client.getRef().getName()),
+          e);
+    }
+    String metadataLocation = null;
+    Reference reference = client.getRef().getReference();
+    try {
+      Content content = client.getApi().getContent().key(key).reference(reference).get().get(key);
+      LOG.debug("Content '{}' at '{}': {}", key, reference, content);
+      if (content == null) {
+        if (currentMetadataLocation() != null) {
+          throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+        }
+      } else {
+        this.icebergView =
+            content
+                .unwrap(IcebergView.class)
+                .orElseThrow(
+                    () -> {
+                      if (content instanceof IcebergTable) {
+                        return new AlreadyExistsException(
+                            "Table with same name already exists: %s in %s", key, reference);
+                      } else {
+                        return new IllegalStateException(
+                            String.format(
+                                "Cannot refresh Iceberg view: Nessie points to a non-Iceberg object for path: %s.",
+                                key));
+                      }
+                    });
+        metadataLocation = icebergView.getMetadataLocation();
+      }
+    } catch (NessieNotFoundException ex) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s in %s", key, reference);
+      }
+    }
+    refreshFromMetadataLocation(metadataLocation, null, 2, l -> loadViewMetadata(l, reference));
+  }
+
+  private ViewMetadata loadViewMetadata(String metadataLocation, Reference reference) {
```

Comment:
does this need to be moved to `NessieUtil` so that it can be used from Trino? (similar to https://github.com/apache/iceberg/pull/7893)